### PR TITLE
Remove nonsensical cast_to_email and cast_to_uri functions

### LINKED ISSUE #3555: Installation creates nonsensical cast functions
## Description

The installation of cast functions currently creates many such functions which make no sense, and can only fail. For example, we create a function that attempts to cast a `bigint` to a `mathesar_types.email`. However, such a conversion is impossible. There's no such thing as a value which is both a valid integer and a valid email address.

## Expected behavior

Only cast functions which make sense should be installed.

The main issue seems to be the constant `db.types.categories.STRING_LIKE_TYPES`. We use this to construct a bunch of textual cast functions in the `db.types.operations.cast.create_textual_casts` function. However, we consider _many_ types (e.g., `mathesar_types.email` to be `STRING_LIKE_TYPES`, and most don't make sense as the target of arbitrary casts from other arbitrary types.

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #3555

**Updated list of valid cast functions for `email` and `uri` types:**
```
mathesar=# \df mathesar_types.cast_to_email
                                  List of functions
     Schema     |     Name      |   Result data type   | Argument data types  | Type 
----------------+---------------+----------------------+----------------------+------
 mathesar_types | cast_to_email | mathesar_types.email | character            | func
 mathesar_types | cast_to_email | mathesar_types.email | character varying    | func
 mathesar_types | cast_to_email | mathesar_types.email | mathesar_types.email | func
 mathesar_types | cast_to_email | mathesar_types.email | text                 | func
(4 rows)

mathesar=# \df mathesar_types.cast_to_uri
                               List of functions
     Schema     |    Name     |  Result data type  | Argument data types | Type 
----------------+-------------+--------------------+---------------------+------
 mathesar_types | cast_to_uri | mathesar_types.uri | character           | func
 mathesar_types | cast_to_uri | mathesar_types.uri | character varying   | func
 mathesar_types | cast_to_uri | mathesar_types.uri | mathesar_types.uri  | func
 mathesar_types | cast_to_uri | mathesar_types.uri | text                | func
(4 rows)

```
## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/types/operations/cast.py [hunk 1]
```diff
@@ -64,11 +63,6 @@ def create_boolean_casts(engine):
     create_cast_functions(PostgresType.BOOLEAN, type_body_map, engine)
 
 
-def create_date_casts(engine):
-    type_body_map = _get_date_type_body_map()
-    create_cast_functions(PostgresType.DATE, type_body_map, engine)
-
-
```
Comment (Anish9901) [inline db/types/operations/cast.py:71]: Since we create identical cast functions in `create_datetime_casts()`, I've removed this function.

Comment (mathemancer) [reply db/types/operations/cast.py:71]: Nice catch.

File: db/types/categories.py [hunk 2]
```diff
@@ -10,8 +10,6 @@
     *STRING_TYPES,
     PostgresType.CHAR,
     PostgresType.NAME,
-    MathesarCustomType.URI,
-    MathesarCustomType.EMAIL,
```
Comment (mathemancer) [inline db/types/categories.py:14]: Removing these with no other changes results in breaking searching for emails and URLs in the record selector.

**Review by mathemancer — CHANGES_REQUESTED** [review]
You need to either:
- restore these, but make a different category for installing casts, or
- Use different logic in the record selector search function (see `db/records/operations/relevance.py`)

**Review by mathemancer — APPROVED** [review]
Okay, this is good to go. Thanks!

**Pull request merged**